doc/ideas/savefileformat.md

```diff
@@ -0,0 +1,446 @@
+
+draft how triggers can be implemented:
+
+event -> conditions -> actions
+
+|           |                                                                                                                           |
+| --------- | ------------------------------------------------------------------------------------------------------------------------- |
+| event     | simple, must be fast calculable (only if true we will look at the conditions), event attributes can be used by conditions |
+| condition | returns a boolen if a speficic gamestate is true, has access @event                                                       |
+| action    | an action which changes the gamestate                                                                                     |
+
+#### 10.1 Event list
+
+| id  | event               | parameter                       | passed to conditions         | info                                                                                 |
+| --- | ------------------- | ------------------------------- | ---------------------------- | ------------------------------------------------------------------------------------ |
+|   0 | unit_dies           | -                               | unit,playerid,teamid         | fired when a unit of player PLAYERID is killed by an enemy(not converted or deleted) |
+|   1 | building_finished   | -                               | building,playerid,teamid     | fired when player PLAYERID has finished building TYPE                                |
+|   2 | building_destroyed  | -                               | building,playerid,teamid     | fired when a building of player PLAYERID is destroyed                                |
+|   4 | unit_created        | -                               | unit,playerid,teamid         | fired when player PLAYERID creates unit UNITTYPE                                     |
+|   5 | intervall           | TIME uint+                      | -                            | fired every INTERVALL milliseconds, but max only once per tick                       |
+|   6 | timer               | TIME uint+                      | -                            | fired once after game is running >= TIME milliseconds                                |
+|   7 | object_in_area      | POLYGON uint+,uint+,uint+,uint+ | unit,playerid,teamid         | fired when an object is in area POLYGON                                              |
+|   8 | tech_researched     | -                               | tech,playerid,teamid         | fired when someone researched TECH                                                   |
+|   9 | unit_conversion     |                                 | unit,playerFrom,playerTo     | fired when a unit is converted and not killed (tech: Heresy)                         |
+|  10 | building_conversion |                                 | building,playerFrom,playerTo | fired when a building is converted and not destroyed (tech: Heresy)                  |
```

Comment:
all this is just stored as keyframes, listing it explicitly will very likely never be done because we can't know our data fields yet, and want to extend them dynamically without pain and backward-compatibility-breaking.